tests/llm/utils/test_case_utils.py

```
@@ -519,3 +531,31 @@ def create_eval_llm(model: str, tracer=None) -> DefaultLLM:
+    to include unique identifiers in prompts to prevent cache hits.
+
+    Example:
+        user_prompt: "Find logs for application {{ env.EVAL_RUN_ID }}"
+
+    This function should be called at test execution time (after before_test runs)
+    so that EVAL_RUN_ID is available.
+    """
+    prompt = test_case.user_prompt
+    if not isinstance(prompt, str):
+        # For list prompts, this should be the specific variant
+        raise ValueError(f"Expected string user_prompt, got {type(prompt)}")
```

Ruff (0.14.14)

[warning] Prefer `TypeError` exception for invalid type

(TRY004)